In-Car Video Creation

larsll opened the pull request:
*Issue #, if available:*

*Description of changes:*


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

larsll commented:
Comments to reviewers - @marcobuss @StevenAskwith @esbjj 
* This is a request for initial review; it has used the techniques used in other features, but is my first attempt at such development.
* It includes #80 and #85, and they should be reviewed / merged first, as the build is partially dependent on it.
* The 500 objects in CloudFormation limit was reached during the development, and some work-arounds like a Shared Role for Lambda functions within a component was introduced to limit the number ofRoles and Policies being introduced.

StevenAskwith commented:
Can you resolve the conflict in "website/public/locales/en/translation.json" please?

I've merged all the preceding PRs

larsll commented:
> Can you resolve the conflict in "website/public/locales/en/translation.json" please?
> 
> I've merged all the preceding PRs

Thank you for the PR, I have rebased and fixed the conflict so that this PR now looks cleaner.

larsll commented:
**Testing** 

- Create a user `Tester` in the testing DREM instance.
- Upload the `AtoZ-CW-Centerline-tracking.tar.gz` model (from the repo) into the account of Tester.
- As admin move the model to your test car.
- If you have the DeepRacer-Custom-Car stack installed you can run inference with the model for a while. or extract the attached file into `/opt/aws/deepracer/logs`.
- As admin:
   - Go to Edit Car page, click Fetch Logs.
   - Under Model Management / Car Logs & Videos - go to the Processing Logs tab and observe your file being processed.
![image](https://github.com/user-attachments/assets/71b2285b-31e8-4276-8a06-b311b6f2d0c3)
   - Once processing completed go to the Assets tab and see that one Bag and one Video asset is created.
![image](https://github.com/user-attachments/assets/1ef4400f-997a-4289-8dec-29490d3ffab2)
- As Tester:
    - Go to Car Logs & Videos and see the same assets being displayed.
![image](https://github.com/user-attachments/assets/27a28f04-3b2e-4bc2-be95-75eee007aed6)
    - Select the video, click download.
    - View the video.
![image](https://github.com/user-attachments/assets/ed44848a-1f94-4552-ada2-80575324af2c)

Bag file: [drpi-1000_20250111-125727.tar.gz](https://github.com/user-attachments/files/18385935/drpi-1000_20250111-125727.tar.gz)
